codespell_lib/_codespell.py

```diff
@@ -361,12 +362,39 @@ def parse_options(args):
                         help='print LINES of leading context')
     parser.add_argument('-C', '--context', type=int, metavar='LINES',
                         help='print LINES of surrounding context')
+    parser.add_argument('--config', type=str,
+                        help='path to config file.')
 
     parser.add_argument('files', nargs='*',
                         help='files or directories to check')
 
+    # Parse command line options.
     options = parser.parse_args(list(args))
 
+    # Load config files and look for ``codespell`` options.
+    cfg_files = ['setup.cfg', '.codespellrc']
+    if options.config:
+        cfg_files.append(options.config)
+    config = configparser.ConfigParser()
+    config.read(cfg_files)
+
+    if config.has_section('codespell'):
+        # Build a "fake" argv list using option name and value.
+        cfg_args = []
+        for key in config['codespell']:
+            # Add option as arg.
+            cfg_args.append("--%s" % key)
+            # If value is blank, skip.
+            val = config['codespell'][key]
+            if val != "":
+                cfg_args.append(val)
+
+        # Parse config file options.
+        options = parser.parse_args(cfg_args)
+
+        # Re-parse command line options to override config.
+        options = parser.parse_args(list(args), namespace=options)
```

Comment:
Rather than re-calling this, would it be better to do:
```
if config.has_section(...):
    ...
    options = parser.parse_args(cfg_args)
else:
    options = argparse.NameSpace()

options = parser.parse_args(list(args), namespace=options)
```
If it's equivalent, it seems simpler/cleaner

Reply:
It only re-parses if `config.has_section()`. The args have to be parsed first (line 372) before checking the config, to see if there is a `--config` option passed. And they have to be re-parsed at the end in order to override the options from the config file (which was the requested behavior). Following your suggestion, you would be re-parsing it twice if there were no config files, which seems pointless.

It's your project though, feel free to re-implement it yourself if you have a better idea.

Reply:
Ahh right, forgot about the need to parse to see the `--config` value. This should be fine then!